aspen/http/resource.py

```diff
@@ -7,25 +8,18 @@ class Static(object):
     """
 
     def __init__(self, request_processor, fspath, raw, media_type):
-        self.request_processor = request_processor
+        assert type(raw) is bytes  # sanity check
         self.raw = raw
         self.media_type = media_type
         if media_type == 'application/json':
-            self.media_type = self.request_processor.media_type_json
+            self.media_type = request_processor.media_type_json
+        if self.media_type.startswith('text/'):  # XXX this is bad
```

Comment:
> What should we do instead?

I wasn't sure (hence a comment instead of a fix), but after b848f839bf92f2e6778529b34c7a8a6203706ed7 this code is only executed when we initialize the resource, not on every request, so we could simply try decoding `raw` with `charset_static` (in which case it wouldn't be purely declarative anymore so we'd switch to `parse.codec`, addressing your earlier comment).